### ISSUE TITLE
AppVMs Remote Desktop (VNC) Issues

### ISSUE THREAD

**Issue by evwwhmxgzmat** [QubesOS/qubes-issues#1153]
I'm trying to control an AppVM remotely using a VNC client/server software. The problem is that when I connect, only a blank screen is shown in VNC viewer. I believe that this is due to the fact that TemplateVMs and AppVMs do not have a desktop on their own, as their desktop functionality is fully integrated with Dom0 desktop GUI.

Is there any way to restore/activate AppVMs own desktop (better if temporarily) to be used as a base for VNC viewer? Hints and suggestions are very welcome... :)

**Comment by unman** [comment]
This might have been better sent to the users mailing list.
Here's one thing you could try:
I'm assuming that you have something like tigervncserver installed and configured on vncVM, and have configured firewall connections to allow access.
Open a terminal in vncVM and start the server: 
systemctl start vncserver@:1.service

On the client, start your viewer and open the connection, entering password if set.
This should give you the "blank screen" you refer to.

Open another term in vncVM, and type
export DISPLAY=:1 (to match the display number you started).
starting xterm should then show a terminal window in the vnc viewer.

Of course, you likely want a window manager of some sort - install ratpoison or stumpWM and start it from vncVM. Or you could even run gnome-session.

**Comment by evwwhmxgzmat** [comment]
Thanks for your help.
I didn't even know that a Qubes users mailing list existed... :(
Next time, I'll post to the right place.

I followed your advice and I created two TemplateVMs:
- fedora-21-vnc, with IP 10.137.2.13 (to use as VNC server)
- fedora-21-vnc-clone-1, with IP 10.137.2.15 (to use as VNC viewer)

I installed tigervnc and tigervnc-server from Fedora 21 repos.
After configuring the firewall to allow connections between the two VMs (as explained in Qubes-os docs) I fired vncserver and vncviewer (see screenshots below).

![vncserver](https://cloud.githubusercontent.com/assets/14013595/9563959/57df9866-4e85-11e5-99f4-4bbacc210bc0.png)
![vncviewer](https://cloud.githubusercontent.com/assets/14013595/9563968/b4f388c8-4e85-11e5-8692-969472767850.png)

The result is shown in the following picture: a blank screen only showning some configuration info and no terminal at all (the menu in the middle is from F8 key).

![blank](https://cloud.githubusercontent.com/assets/14013595/9563971/d62ce390-4e85-11e5-96fe-08456a27c227.png)

However, during my experiments I discovered something weird. After starting the VNC server in fedora-21-vnc, if I run a second terminal it will not appear, and the icon below the cursor arrow will keep bouncing for a while.
Now, if during this time I run vncviewer in fedora-21-vnc-clone-1, I do get a terminal and I can fire gnome-session, which starts succsessfully as shown in the following screenshot.

![gnome-session-bis](https://cloud.githubusercontent.com/assets/14013595/9563973/e999dc08-4e85-11e5-8fa6-a2259b36e0a3.png)

Why does the terminal in the remote session starts only in that particular situation? Is that a bug?
